p2p/simulations/network.go

```diff
@@ -168,15 +183,57 @@ func NewNetworkController(net NetworkControl, nodesController *ResourceControlle
 		},
 	)
 
-	self.SetResource("nodes", nodesController)
+	self.SetResource("nodes", NewNodesController(network, mocker))
 	self.SetResource("debug", NewDebugController(journal))
 	self.SetResource("mockevents", NewMockersController(eventer, conf.DefaultMockerConfig))
 	self.SetResource("journals", NewJournalPlayersController(eventer))
 
 	return &NetworkController{ResourceController: self, events: net.Events()}
 }
 
-func NewNodesController(net *Network) *ResourceController {
+func NewNodesController(net *Network, mocker *NodeMocker) *ResourceController {
+  ids     := mocker.CreateMockNodes(net.Config().DefaultMockerConfig.NodeCount)
+  net.init(ids)
+	nc      := NewResourceContoller(
+		&ResourceHandlers{
+			//GET /<networkId>/nodes  -- returns all nodes' kademlia table
```

Comment:
we can always use `network.GetNodeAdapter(id).(*SimNode).hive` now so we dont need the hives member
not sure we need to define another costom Network even